### ISSUE TITLE
Using app.add_css_stylesheet() in conf.py

### ISSUE THREAD

**Issue by diego-plan9** [jupyterhub/oauthenticator#460]
When attempting to build the documentation using latest sphinx (`4.2.0`) (related to an effort of packaging `oauthenticator` into Debian), an error is raised due to:
```
app.add_stylesheet('custom.css')
```
at [`conf.py`](https://github.com/jupyterhub/oauthenticator/blob/master/docs/source/conf.py#L130).

It seems `.add_stylesheet()` has been technically deprecated since quite a while (`1.8`), although not really remove from sphinx until recently - while it might get reintroduced and supported until 6.0 (https://github.com/sphinx-doc/sphinx/issues/9683), would it make sense to replace it with its `.add_css_stylesheet()` successor, for some future-proofing? Actually, I could not trace the referenced `custom.css` - not sure if an alternative would be to simply drop the line altogether.

**Comment by consideRatio** [comment]
Absolutely, sounds very reasonable! Let's transition to `add_css_stylesheet` and make us require sphinx >=2 if we don't already do.

**Comment by tk0miya** [comment]
>would it make sense to replace it with its .add_css_stylesheet() successor, for some future-proofing? 

I think it's better to use the `html_css_file` configuration instead of the `add_css_file()` API (if you still need to install custom CSS).
https://www.sphinx-doc.org/en/master/usage/configuration.html#confval-html_css_files
